src/makeCalculator.js

```diff
@@ -37,7 +37,45 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    operate,
+    add,
+    subtract,
+    multiply,
+    divide,
+    reset,
+    result: 0,
+  };
+
+  function operate(operation, number = 0) {
+    operation.call(this, number);
+
+    return this;
+  }
+
+  function add(number) {
+    this.result += number;
+  }
+
+  function subtract(number) {
+    this.result -= number;
+  }
+
+  function multiply(number) {
+    this.result *= number;
+  }
+
+  function divide(number) {
+    this.result /= number;
```

Comment:
There is no check for division by zero. If number passed as argument is zero, it will lead to an error as division by zero is undefined.